tests/test_cli.py

```diff
@@ -91,13 +353,54 @@ def test_history_uid():
     assert "Show this message and exit" in result.output
 
 
+def test_history_view_success(caplog):
+    runner = CliRunner()
+    dummy_uid = str(uuid.uuid4())  # make dummy UID
+    # add a dummy entry
+    cmd.add(
+        {
+            "uid": dummy_uid,
+            "command": "dummy --test",
+            "timestamp": "2025-04-17T12:00:00Z",
+        }
+    )
+    result = runner.invoke(cli, ["history", "uid", dummy_uid])
+    assert result.exit_code == 0
+    assert "INFO: Execution UID: {dummy_uid}" in caplog.text
+    assert "INFO: Command:" in caplog.text
+    assert "INFO: Logs" in caplog.text
+
+
+# testing failure of history uid command
+def test_history_view_success(caplog):
+    runner = CliRunner()
+    result = runner.invoke(cli, ["history", "uid", "fail-uid"])
+    assert result.exit_code == 1
+    assert "ERROR: No log entry found for UID:" in caplog.text
+
+
 def test_history_view():
     runner = CliRunner()
     result = runner.invoke(cli, ["history", "view", "--help"])
     assert result.exit_code == 0
     assert "Show this message and exit" in result.output
 
 
+def test_history_view_success(caplog):
+    runner = CliRunner()
+    result = runner.invoke(cli, ["history", "view"])
+    assert result.exit_code == 0
+    assert "recent log entries:" in caplog.text
+
+
+# testing success of history view (with specified line number)
+def test_history_view_success(caplog):
```

Comment:
Tests shouldn't have the same name and it is ok for these names to be very verbose (i.e. long). So you can call this `test_history_view_with_n_option_success`, etc